core/src/main/java/org/elasticsearch/gateway/Gateway.java

```diff
@@ -129,7 +140,17 @@ public void performStateRecovery(final GatewayStateRecoveredListener listener) t
                 if (electedIndexMetaData != null) {
                     if (indexMetaDataCount < requiredAllocation) {
                         logger.debug("[{}] found [{}], required [{}], not adding", index, indexMetaDataCount, requiredAllocation);
+                    } // TODO if this logging statement is correct then we are missing an else here
```

Comment:
that logging message is useless - it is also relevant for the case where the state was fetched from enough nodes, but some index isn't present on requiredAllocation - which is extremely rare. IMO we should just remove all index level "selection". We elect a global state and that's it. I'll this as a follow up.